client/src/app/site/agenda/services/agenda-repository.service.ts

```diff
@@ -45,27 +49,120 @@ export class AgendaRepositoryService extends BaseRepository<ViewItem, Item> {
+     * Might need another repo
+     * @param id {@link User} id of the new speaker
+     * @param agenda the target agenda item
+     */
+    public addSpeaker(id: number, agenda: Item): Observable<any> {
+        const restUrl = `rest/agenda/item/${agenda.id}/manage_speaker/`;
+        return this.httpService.create(restUrl, { user: id });
+    }
+
+    /**
+     * Sets the given speaker ID to Speak
+     * @param id the speakers id
+     * @param agenda the target agenda item
+     */
+    public startSpeaker(id: number, agenda: Item): void {
+        const restUrl = `rest/agenda/item/${agenda.id}/speak/`;
+
+        this.httpService.update(restUrl, { speaker: id }, HTTPMethod.PUT).subscribe(response => {
+            console.log('Set a new speaker? ', response);
+        });
+    }
+
+    /**
+     * Stops the current speaker
+     * @param agenda the target agenda item
+     */
+    public stopSpeaker(agenda: Item): void {
+        const restUrl = `rest/agenda/item/${agenda.id}/speak/`;
+
+        this.httpService.delete(restUrl, {}).subscribe(response => {
+            console.log('Stopped a speaker? ', response);
+        });
+    }
+
+    /**
+     * Marks the current speaker
+     * @param id {@link User} id of the new speaker
+     * @param doMark determine if the user was marked or not
+     * @param agenda the target agenda item
+     */
+    public markSpeaker(id: number, mark: boolean, agenda: Item): void {
+        const restUrl = `rest/agenda/item/${agenda.id}/manage_speaker/`;
+
+        this.httpService.update(restUrl, { user: id, marked: mark }, HTTPMethod.PATCH).subscribe(response => {
+            console.log('Mark a speaker ? ', response);
+        });
+    }
+
+    /**
+     * Deletes the given speaker for the agenda
+     * @param id the speakers id
+     * @param agenda the target agenda item
+     */
+    public deleteSpeaker(agenda: Item, id?: number): void {
+        const restUrl = `rest/agenda/item/${agenda.id}/manage_speaker/`;
+
+        /**
+         * FIXME: Angular does not support to have a request payload in delete requests
+         */
+        this.httpService.delete(restUrl, { speaker: id }).subscribe(response => {
+            console.log('a speaker was deleted: ', response);
+        });
+    }
+
+    /**
+     * Posts an (manually) sorted speaker list to the server
+     * @param ids array of speaker id numbers
+     * @param Item the target agenda item
+     */
+    public sortSpeakers(ids: number[], agenda: Item): void {
+        const restUrl = `rest/agenda/item/${agenda.id}/sort_speakers/`;
+
+        this.httpService.create(restUrl, { speakers: ids }).subscribe(response => {
+            console.log('Sorted the list of speakers? ', response);
+        });
+    }
+
+    /**
+     * @ignore
      */
     public update(item: Partial<Item>, viewUser: ViewItem): Observable<Item> {
         return null;
     }
 
     /**
      * @ignore
-     *
-     * TODO: used over not-yet-existing detail view
```

Comment:
Do not delete this, if it's not implemented ;)

Reply:
Ok, add a comment then, that they should not be used (and may just throw an error there, like in the config repo).